controllers/datadogagent/feature/apm/feature.go

```diff
@@ -363,6 +392,19 @@ func (f *apmFeature) manageNodeAgent(agentContainerName apicommonv1.AgentContain
 	}
 	managers.Port().AddPortToContainer(agentContainerName, apmPort)
 
+	// APM SSI Language Detection
+	if f.shouldEnableLanguageDetection() {
+		managers.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  apicommon.DDLanguageDetectionEnabled,
+			Value: "true",
+		})
+
+		managers.EnvVar().AddEnvVarToContainer(apicommonv1.ProcessAgentContainerName, &corev1.EnvVar{
```

Comment:
> Is the process agent required for this feature?

Yes process agent is required for this feature.
The process agent runs the process check which detects running processes and their respective languages. 
The core agent then gets these collected processes via the remote workloadmeta collector, and leverage these data to provide the rest of the feature execution.
I think I should update the list of required components to include the process agent.

> Also out of curiosity, would this be affected by the DD_PROCESS_CONFIG_RUN_IN_CORE_AGENT_ENABLED=true setting?

Could you please clarify your question, I don't get it.

Reply:
Since `DD_PROCESS_CONFIG_RUN_IN_CORE_AGENT_ENABLED=true` would make features like live processes, live containers, and process discovery run in the core agent container instead of the process agent container, I was wondering if `DD_PROCESS_CONFIG_PROCESS_COLLECTION_ENABLED` would also need to be moved to the core agent based on `DD_PROCESS_CONFIG_RUN_IN_CORE_AGENT_ENABLED`. Live processes can [add the env var to the core agent](https://github.com/DataDog/datadog-operator/blob/7667a52a94b75e2a30af4411deaa445819a1bc4e/controllers/datadogagent/feature/liveprocess/feature.go#L134), so if this feature is also enabled, I think we could have `DD_PROCESS_CONFIG_PROCESS_COLLECTION_ENABLED=true` in both core agent and process agent containers. Would that cause an issue?

Reply:
I don't think having `DD_PROCESS_CONFIG_COLLECTION_ENABLED=true` in both the core agent and the process agent can cause an issue, as long as process collection is still done correctly and process data are being (finally) populated in the workloadmeta store of the core agent (regardless of whether the process check was executed in the process agent or in the core agent).